AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs

```diff
@@ -24,6 +24,28 @@ public class AzureCSharpCodeNamer : CSharpCodeNamer
             // Do nothing
         }
 
+        public AzureCSharpCodeNamer(Settings settings)
+        {
+            if (settings != null && settings.CustomSettings != null)
```

Comment:
Is this an error condition for your object? Is it interesting to log about whether or not AzureCSharpCodeNamer got a valid settings object and/or took action on the CustomSetting?